### ISSUE TITLE
[Issue] HTTP Endpoint not working with Environment Variables

### ISSUE THREAD

**Issue by fjunior87** [wso2/micro-integrator#1544]
**Description:**
For HTTP Endpoint it is not being able to use Environment Variables for the URI.
In the documentation[1], it states for HTTP endpoint we can inject in the uri property, but we don't have that property for this type of endpoint. And when I try with uri-template it does not work and does not resolve the environment variable.

But it does work correctly with Address endpoint. 

I tried injecting in a variable like uri.var.epUrl, but as described in [2] the scope env is not working in MI as it is in EI 6.6.0.

[1] https://ei.docs.wso2.com/en/latest/micro-integrator/develop/injecting-parameters/
[2]https://github.com/wso2/micro-integrator/issues/1542

**Suggested Labels:**
issue
**Suggested Assignees:**

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**
export ACCOUNT_BASE_URL=http://localhost:8290/services/MyService/myResource
create an endpoint like below and try to use it:
```
<call>
    <endpoint>
        <http method="get" uri-template="$SYSTEM:ACCOUNT_BASE_URL">
            <suspendOnFailure>
                <initialDuration>-1</initialDuration>
                <progressionFactor>1</progressionFactor>
            </suspendOnFailure>
            <markForSuspension>
                <retriesBeforeSuspension>0</retriesBeforeSuspension>
            </markForSuspension>
        </http>
    </endpoint>
</call>

```
**Related Issues:**
1542

**Comment by arunans23** [comment]
Tested with following sample in WSO2 MI 1.2.0-alpha. Could not reproduce.
```
<?xml version="1.0" encoding="UTF-8"?>
<api context="/mocky" name="Mocky" xmlns="http://ws.apache.org/ns/synapse">
    <resource methods="GET">
        <inSequence>
            <call>
                <endpoint>
                    <http method="get" uri-template="$SYSTEM:MOCKY_EP">
                        <suspendOnFailure>
                            <initialDuration>-1</initialDuration>
                            <progressionFactor>-1</progressionFactor>
                            <maximumDuration>0</maximumDuration>
                        </suspendOnFailure>
                        <markForSuspension>
                            <retriesBeforeSuspension>0</retriesBeforeSuspension>
                        </markForSuspension>
                    </http>
                </endpoint>
            </call>
            <respond/>
        </inSequence>
        <outSequence/>
        <faultSequence/>
    </resource>
</api>

```

**Comment by fjunior87** [comment]
Hi @arunans23 , Sorry, but I have tested with MI 1.1.0. If this is fixed in the newer versions then it is fine and we can close the issue. But it would be good if we could have that as an update for previous version.

Cheers

**Comment by arunans23** [comment]
@fjunior87, It seems to be fixed in the update for 1.1.0 as well.

**Comment by arunans23** [comment]
Closing the issue, since it is fixed in the latest version.
